src/main/scala/components/memory/memory-noncombin-ports.scala

```diff
@@ -17,11 +17,22 @@ class OutstandingReq extends Bundle {
 
 /**
  * The instruction memory port. Since both the combinational and noncombinational instruction ports just issue
- * read requests in the same way both ports have the same implementation
+ * read requests in the same way both ports share the same implementation
  *
  * The I/O for this module is defined in [[IMemPortIO]].
  */
 class INonCombinMemPort extends ICombinMemPort {
+  // To fake combinational writes
+  val imemBusy  = RegInit (false.B)
+  imemBusy := DontCare
+
+  when (io.pipeline.valid) {
+    imemBusy := true.B
+  } .elsewhen (io.bus.response.valid) {
+    imemBusy := false.B
+  }
+
+  io.pipeline.good := (~ imemBusy || io.bus.response.valid)
```

Comment:
I don't quite follow this logic. imemBusy will always be false when io.bus.response.valid is true, right? Also, it will be dontcare otherwise?

Reply:
When the non-combinational instruction port sends a request it will set imemBusy to high. A number of cycles later the memory bus will respond with true on the valid signal, thus implicitly indicating that it is ready for another request, but on that specific cycle ``imemBusy`` will also be high and needs one more cycle to be updated to false. 

So I OR'd the two signals together in this case. The memory will take one extra cycle to perform operations otherwise if we just pay attention to ``imemBusy``.